### ISSUE TITLE
Assuming netcat exists can cause problems

### ISSUE THREAD

**Issue by ericfranz** [OSC/ood_core#153]
See: https://github.com/OSC/ood_core/blob/a87f59ea9d24cdcbe505c9ab859bcb120bae9951/lib/ood_core/batch_connect/template.rb#L120-L126

The assumption here is that netcat is available. If it is not (because the command doesn't exist or the user is not allowed to execute the command) the result will be nonzero but will be reported as the port not yet being in use.

If we are going to use netcat, we should check for its existence and handle the error case that, then use fallback solutions.

Or not use netcat at all. See https://listsprd.osu.edu/pipermail/ood-users/2018-October/000269.html for discussion and some options and Trey's comment:

>I’ve used this trick in the past to check if ports are available when bootstrapping MySQL instances in jobs that shared the same node:
> 
>    if (: < /dev/tcp/127.0.0.1/${MYSQL_PORT}) 2>/dev/null; then
>        echo "MySQL port ${MYSQL_PORT} already taken."
>        return 1
>    fi
>

**Comment by ericfranz** [comment]
On solution if `lsof` is available on the compute nodes to add to the before.sh.erb before `find_port` is executed:

```bash
# override
port_used () {
  local port="${1#*:}"
  local host=$((expr "${1}" : '\\(.*\\):' || echo "localhost") | awk 'END{print $NF}')
  lsof -i :"${port}" &> /dev/null
}
export -f port_used
```

**Comment by ericfranz** [comment]
If there is a reliable way to check if the pseudo-device /dev/tcp is available, we would check if that exists first, then use it if it does; if not then try something else.

**Comment by ericfranz** [comment]
Addressed by #159
